File: Quivi.Guests.Api/Dtos/Requests/Orders/CreateOrderRequest.cs
namespace Quivi.Guests.Api.Dtos.Requests.Orders
{
    public class CreateOrderRequest : ARequest
    {
        public required string ChannelId { get; init; }
        public required IEnumerable<OrderItem> Items { get; init; }
    }

    public class OrderItem : BaseOrderItem
    {
        public IEnumerable<OrderItemModifierGroup>? ModifierGroups { get; init; }
    }

    public class BaseOrderItem
    {
        public required string MenuItemId { get; init; }
        public int Quantity { get; init; }
    }

    public class OrderItemModifierGroup
    {
        public required string ModifierId { get; init; }
        public required IEnumerable<BaseOrderItem> SelectedOptions { get; init; }
    }
}

File: quivi.guests.react.app/src/hooks/api/Dtos/orders/CreateOrderRequest.ts
export interface CreateOrderBaseItem {
    readonly menuItemId: string;
    readonly quantity: number;
}

export interface CreateOrderItemModifier {
    readonly modifierId: string;
    readonly selectedOptions: CreateOrderBaseItem[];
}

export interface CreateOrderItem extends CreateOrderBaseItem {
    readonly modifierGroups: CreateOrderItemModifier[];
}

export interface CreateOrderRequest {
    readonly merchantId: string;
    readonly channelId: string;
    readonly payLater: boolean;
    readonly items: CreateOrderItem[];
}